File: l1/z1/LiczbyPierwsze.java
import java.util.ArrayList;

public class LiczbyPierwsze {
	private ArrayList<Integer> prime_nums = new ArrayList<Integer>();

	LiczbyPierwsze(int n) {
		prime_nums.add(2);
		for (int i = 2; i <= n; i++) {
			boolean is_ok = true;

			for (Integer divider : prime_nums) {
				if (i % divider == 0) {
					is_ok = false;
				}
			}

			if (is_ok){
				prime_nums.add(i);
			}
		}
	}
	
	public int liczba(int m){
		return prime_nums.get(m);
	}
}

File: l1/z2/LiczbyPierwsze.cpp
#include "LiczbyPierwszeHeader.hpp"
#include <vector>

LiczbyPierwsze::LiczbyPierwsze(int n) { 
	LiczbyPierwsze::prime_nums.push_back(2);
	for (int i = 2; i <= n; i++) {
		bool is_ok = true;

		for (int index = 0; index < prime_nums.size(); index++){
			if (i % prime_nums[index] == 0){
				is_ok = false;
			}
		}

		if (is_ok){
			prime_nums.push_back(i);
		}
	}
}

int LiczbyPierwsze::liczba(int m){
	return prime_nums[m];
}

int LiczbyPierwsze::count(){
	return prime_nums.size();
}
